In [ ]:
# import pandas as pd
# from google.transit import gtfs_realtime_pb2
# from google.protobuf.json_format import MessageToDict
# import requests
# import datetime
# import json

# # Wczytanie danych GTFS-RT
# feed = gtfs_realtime_pb2.FeedMessage()
# response = requests.get('https://gtfs.ztp.krakow.pl/VehiclePositions_A.pb')
# feed.ParseFromString(response.content)


# entity_dict = MessageToDict(feed.entity[0], preserving_proto_field_name=True)

# # Ładne wyświetlenie JSON
# print(json.dumps(entity_dict, indent=2))
# # Lista na dane
# vehicles = []

# # Przetwarzanie danych
# for entity in feed.entity:
#     if entity.HasField('vehicle'):
#         vehicle = entity.vehicle

#         vehicles.append({
#             'vehicle_id': vehicle.vehicle.id,
#             'trip_id': vehicle.trip.trip_id,
#             'route_id': vehicle.trip.route_id,
#             'latitude': vehicle.position.latitude,
#             'longitude': vehicle.position.longitude,
#             'bearing': vehicle.position.bearing,
#             'speed': vehicle.position.speed,
#             'timestamp': datetime.datetime.fromtimestamp(vehicle.timestamp),
#             'stop_id':vehicle.stop_id,
#             'stop_sequence':vehicle.current_stop_sequence,
#             'current_date': datetime.date.today(),
#             'day': datetime.date.today().isoweekday()
#         })

# # Konwersja do DataFrame
# df = pd.DataFrame(vehicles)

# # Zapis do CSV (opcjonalnie)
# df.to_csv('vehicle_positions.csv', index=False)

# # Podgląd
# print(df.head())

In [ ]:
# import pandas as pd
# import opoznienie
# from google.transit import gtfs_realtime_pb2
# import requests
# import datetime
# import os

# ref_df=pd.read_csv('../data/rozklad/stop_times.txt',delimiter=',')
# positions_df=pd.read_csv('../data/positions/vehicle_positions_2025-04-30_004744.csv')

# merged = pd.merge(positions_df,ref_df[['trip_id', 'stop_id', 'departure_time','stop_sequence']],on=['trip_id', 'stop_id','stop_sequence'],how='left')

# merged['timestamp'] = pd.to_datetime(merged['timestamp'])


# merged['planned_departure'] = merged.apply(
#     lambda row: opoznienie.parse_departure_time_final(row['departure_time'], row['timestamp']) if pd.notnull(row['departure_time']) else pd.NaT,
#     axis=1
# )


# # Obliczanie opóźnienia (w minutach)
# merged['delay_sec'] = (merged['timestamp'] - merged['planned_departure']).dt.total_seconds()

# merged.head()

In [ ]:
# from google.transit import gtfs_realtime_pb2
# from google.protobuf.json_format import MessageToDict
# import requests
# import json

# # Wczytanie danych GTFS-RT
# feed = gtfs_realtime_pb2.FeedMessage()
# response = requests.get('https://gtfs.ztp.krakow.pl/TripUpdates_A.pb')
# feed.ParseFromString(response.content)

# entity_dict = MessageToDict(feed.entity[0], preserving_proto_field_name=True)

# # Ładne wyświetlenie JSON
# print(json.dumps(entity_dict, indent=2))

# # Lista na dane
# vehicles = []

# # Przetwarzanie danych
# for entity in feed.entity:
#     if entity.HasField('vehicle'):
#         vehicle = entity.vehicle

#         vehicles.append({
#             'trip_id': vehicle.trip_update.id,
#             ''
#         })

# # Konwersja do DataFrame
# df = pd.DataFrame(vehicles)

# # Zapis do CSV (opcjonalnie)
# df.to_csv('vehicle_positions.csv', index=False)

# # Podgląd
# print(df.head())

In [ ]:
# import pandas as pd
# import opoznienie
# from google.transit import gtfs_realtime_pb2
# import requests
# import datetime
# import os

# test=pd.DataFrame()
# print(test.empty)


# delays=pd.read_csv('../data/opoznienia/delays.csv')


# #print(delays.axes)

# delays_new=delays[['vehicle_id','trip_id','stop_sequence','current_status','day', 'planned_departure','timestamp','delay_sec']]
# delays_new['delay_min']=delays_new['delay_sec']/60

# delays_new.head()


In [ ]:
#delays_new.loc[delays_new['vehicle_id']==2]


In [ ]:
import pandas as pd
from google.transit import gtfs_realtime_pb2
from google.protobuf.json_format import MessageToDict
import requests
import datetime
import json
import opoznienie
import os

delays=pd.read_csv('../data/data_08_05_25-15_05_25/data/opoznienia/delays_short.csv')
delays=delays.drop_duplicates()
delays=delays.dropna()


In [ ]:
delays.loc[delays['delay_sec']<=-600]
delays = delays[delays['delay_sec'] > -600]


In [ ]:
delays

In [ ]:
#delays_long=delays.loc[delays['delay_sec']>=5200]

# na razie limit od -600 do 5200 sekund opoznienia

# delays_long_unique = delays_long.drop_duplicates(subset=['vehicle_id'])

# delays_long_unique

In [ ]:
delays_long=delays[delays['delay_sec']<5200]

delays_long
#delays_long_unique = delays_long.drop_duplicates(subset=['vehicle_id'])

#delays_long_unique


In [ ]:
trips = pd.read_csv("../data/rozklad/trips.txt")
routes = pd.read_csv("../data/rozklad/routes.txt")

trips2=pd.read_csv("../data/rozklad/sroda_08_10/trips.txt")
routes2=pd.read_csv("../data/rozklad/sroda_08_10/routes.txt")

merged=delays_long.merge(trips[['trip_id', 'route_id']], on='trip_id', how='left')
merged = merged.merge(routes[['route_id', 'route_short_name']], on='route_id', how='left')
merged = merged.drop(columns=['route_id'])

merged2=delays_long.merge(trips2[['trip_id', 'route_id']], on='trip_id', how='left')
merged2 = merged2.merge(routes2[['route_id', 'route_short_name']], on='route_id', how='left')
merged2=merged2.dropna()
merged=merged.dropna()

# nie dziala

combined_merged = pd.concat([merged, merged2], ignore_index=True)
combined_merged = combined_merged.drop_duplicates()


combined_merged['route_short_name']=combined_merged['route_short_name'].astype(int)
combined_merged['route_name']=combined_merged['route_short_name']

combined_merged.drop(columns='route_short_name')

,vehicle_id,trip_id,stop_sequence,current_status,day,planned_departure,timestamp,delay_sec,delay_min,route_id,route_name
0,4,block_1890_trip_14_service_5,1,1,4,2025-05-08 16:55:00,2025-05-08 16:48:12,-408.0,-6.800000,NaN,179
1,11,block_1794_trip_11_service_5,1,1,4,2025-05-08 16:48:00,2025-05-08 16:47:58,-2.0,-0.033333,NaN,144
2,15,block_1800_trip_10_service_5,30,1,4,2025-05-08 16:40:00,2025-05-08 16:48:13,493.0,8.216667,NaN,144
3,36,block_1893_trip_13_service_5,10,1,4,2025-05-08 16:42:00,2025-05-08 16:48:11,371.0,6.183333,NaN,179
4,38,block_2181_trip_12_service_5,12,1,4,2025-05-08 16:37:00,2025-05-08 16:48:15,675.0,11.250000,NaN,503
...,...,...,...,...,...,...,...,...,...,...,...
1161743,530,block_2226_trip_2_service_5,5,1,4,2025-05-15 23:31:00,2025-05-15 23:33:59,179.0,2.983333,route_606,179
1161744,557,block_2227_trip_4_service_5,19,1,4,2025-05-15 23:31:00,2025-05-15 23:33:55,175.0,2.916667,route_143,611
1161745,559,block_1781_trip_20_service_5,1,1,4,2025-05-15 23:39:00,2025-05-15 23:33:56,-304.0,-5.066667,route_580,134
1161746,584,block_2245_trip_5_service_5,1,1,4,2025-05-15 23:35:00,2025-05-15 23:33:50,-70.0,-1.166667,route_666,722


In [ ]:

merged[merged['stop_sequence']==10]

In [ ]:
merged[merged['route_name']==290]

In [ ]:
import seaborn as sns
import matplotlib
import matplotlib.pyplot as plt
avg_delays_by_route_stop1=merged[merged['stop_sequence']==2]

avg_delays_by_route = avg_delays_by_route_stop1.groupby('route_name')['delay_min'].median().reset_index()
order=avg_delays_by_route.sort_values('delay_min',ascending=False)['route_name']
plt.figure(figsize=(15,6))
sns.barplot(data=avg_delays_by_route, x='route_name', y='delay_min',order=order[0:30])
plt.xlabel('Route Name')
plt.ylabel('Average Delay (min)')
plt.title('Average Delays (min) by Route, stop no. 2')
plt.xticks(rotation=90)
plt.tight_layout()
plt.show()

In [ ]:
avg_delays_by_route_stop10=merged[merged['stop_sequence']==10]

avg_delays_by_route = avg_delays_by_route_stop10.groupby('route_name')['delay_min'].median().reset_index()
order=avg_delays_by_route.sort_values('delay_min',ascending=False)['route_name']
plt.figure(figsize=(15,6))
sns.barplot(data=avg_delays_by_route, x='route_name', y='delay_min',order=order[0:30])
plt.xlabel('Route Name')
plt.ylabel('Average Delay (min)')
plt.title('Average Delays (min) by Route, stop no. 10')
plt.xticks(rotation=90)
plt.tight_layout()
plt.show()

In [ ]:
avg_delays_by_route_stop15=merged[merged['stop_sequence']==15]

avg_delays_by_route = avg_delays_by_route_stop15.groupby('route_name')['delay_min'].median().reset_index()
avg_delays_by_route = avg_delays_by_route.sort_values(by='delay_min', ascending=False)

order=avg_delays_by_route.sort_values('delay_min',ascending=False)['route_name']

plt.figure(figsize=(15,6))
sns.barplot(data=avg_delays_by_route,
            x='route_name',
            y='delay_min',
            order=order[0:30])
plt.xlabel('Route Name')
plt.ylabel('Average Delay (min)')
plt.title('median Delays (min) by Route, stop no. 15')
plt.xticks(rotation=90)
plt.tight_layout()
plt.show()

In [ ]:
avg_delays_route433=merged[merged['route_name']==433]

avg_delays_by_route = avg_delays_route433.groupby('stop_sequence')['delay_min'].median().reset_index()
avg_delays_by_route = avg_delays_by_route.sort_values(by='delay_min', ascending=False)

order=avg_delays_by_route.sort_values('delay_min',ascending=False)['stop_sequence']

plt.figure(figsize=(15,6))
sns.barplot(data=avg_delays_by_route,
            x='stop_sequence',
            y='delay_min')
plt.xlabel('stop sequence')
plt.ylabel('Average Delay (min)')
plt.title('median Delays (min) by stop sequence, line no. 433')
plt.xticks(rotation=90)
plt.tight_layout()
plt.show()

In [ ]:
avg_delays_route433=merged[merged['route_name']==173]

avg_delays_by_route = avg_delays_route433.groupby('stop_sequence')['delay_min'].median().reset_index()
avg_delays_by_route = avg_delays_by_route.sort_values(by='delay_min', ascending=False)

order=avg_delays_by_route.sort_values('delay_min',ascending=False)['stop_sequence']

plt.figure(figsize=(15,6))
sns.barplot(data=avg_delays_by_route,
            x='stop_sequence',
            y='delay_min')
plt.xlabel('stop sequence')
plt.ylabel('median Delay (min)')
plt.title('median Delays (min) by stop sequence, line no. 173')
plt.xticks(rotation=90)
plt.tight_layout()
plt.show()

In [ ]:
merged[(merged['route_name'] == 173) & (merged['stop_sequence'] == 28)]


linie       przystanki
424, 502,   Rondo Mogilskie Opera 02
152, 124,
            
152, 424,   Rondo Młyńskie
124, 502,

139, 501,
503, 179,
159, 307,
511, 304,
192, 513,
199, 301,
169, 469,   Nowy Kleparz 04


139, 152,
192, 513,
179, 192,
501, 199,
301, 304,
169,          Plac Inwalidów